### ISSUE TITLE
edd_downloads_list_before hook output buffer issue

### ISSUE THREAD

**Issue by davidsherlock** [easydigitaldownloads/easy-digital-downloads#6777]
### Expected behavior

I built a function to add `header_title` and `header_subtitle` header support to the downloads `shortcode` attributes. I want this to appear outside of the `edd_downloads_list` div, so I used the `edd_downloads_list_before` action hook to achieve this.

When multiple downloads shortcodes were on the page, the titles did not match up properly due to outputting buffering. This is due to the fact the action hook is outside of `ob_start`. When moved within the conditional statement, the code (and hook) work as expected.

I can see the reasoning for both situations, but propose we either relocate the hook or add an additional hook just below the conditional statement.

### Actual behavior
The hook results in unexpected HTML formatting.

**Comment by cklosowski** [comment]
Yeah we'll need to keep the output buffering in place, as it stops errors from making it to the front end and breaking functionality in the event of an error...but we could look at how to better wrap those with the hooks.
